src/d3a/d3a_core/sim_results/endpoint_buffer.py

```diff
@@ -160,6 +161,9 @@ def _populate_core_stats(self, area):
         if hasattr(area.current_market, 'trades'):
             for trade in area.current_market.trades:
                 core_stats_dict['trades'].append(trade.serializable_dict())
+        if getattr(area, 'strategy', None) is None and area.current_market is not None:
+            core_stats_dict['grid_fee_constant'] = area.current_market.fee_class.grid_fee_rate \
+                if isinstance(area.current_market.fee_class, ConstantGridFees) else None
```

Comment:
This works but looks very counter-intuitive to me to use introspection in order to find out the area fee type. I would rather expose an extra property method in the Area class (have a look at Area._set_grid_fees, you can either add an extra argument in order to save the fee type, or deduce the fee type by None-checking self.grid_fee_constant and self.grid_fee_percentage)